### ISSUE TITLE
Warnings when compiling on Ubuntu 16.04 with gcc 5.3.1

### ISSUE THREAD

**Issue by spficklin** [SystemsGenetics/ACE#1]
Ace will compile for me on Ubuntu 16.04 but I get the following warnings:

```
Building object ../build/ace.o
Building object ../build/analyticplugin.o
Building object ../build/clcommandqueue.o
clcommandqueue.cpp: In member function ‘void AccelCompEng::CLCommandQueue::init(cl_context, cl_device_id)’:
clcommandqueue.cpp:20:10: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
    _id = clCreateCommandQueue(cid,did,0x0,&err);
          ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1359:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^
clcommandqueue.cpp:20:10: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
    _id = clCreateCommandQueue(cid,did,0x0,&err);
          ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1359:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^
clcommandqueue.cpp:20:47: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
    _id = clCreateCommandQueue(cid,did,0x0,&err);
                                               ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1359:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^
clcommandqueue.cpp: In member function ‘AccelCompEng::CLEvent AccelCompEng::CLCommandQueue::add_task(AccelCompEng::CLKernel&)’:
clcommandqueue.cpp:33:10: warning: ‘cl_int clEnqueueTask(cl_command_queue, cl_kernel, cl_uint, _cl_event* const*, _cl_event**)’ is deprecated [-Wdeprecated-declarations]
    err = clEnqueueTask(_id,kernel._id,0,NULL,&ret);
          ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1373:1: note: declared here
 clEnqueueTask(cl_command_queue  /* command_queue */,
 ^
clcommandqueue.cpp:33:10: warning: ‘cl_int clEnqueueTask(cl_command_queue, cl_kernel, cl_uint, _cl_event* const*, _cl_event**)’ is deprecated [-Wdeprecated-declarations]
    err = clEnqueueTask(_id,kernel._id,0,NULL,&ret);
          ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1373:1: note: declared here
 clEnqueueTask(cl_command_queue  /* command_queue */,
 ^
clcommandqueue.cpp:33:50: warning: ‘cl_int clEnqueueTask(cl_command_queue, cl_kernel, cl_uint, _cl_event* const*, _cl_event**)’ is deprecated [-Wdeprecated-declarations]
    err = clEnqueueTask(_id,kernel._id,0,NULL,&ret);
                                                  ^
In file included from clcommandqueue.h:3:0,
                 from clcommandqueue.cpp:1:
/usr/include/CL/cl.h:1373:1: note: declared here
 clEnqueueTask(cl_command_queue  /* command_queue */,
 ^
Building object ../build/clcontext.o
Building object ../build/cldevice.o
Building object ../build/cldevlist.o
Building object ../build/clevent.o
Building object ../build/clkernel.o
Building object ../build/clprogram.o
Building object ../build/console.o
Building object ../build/datamap.o
Building object ../build/exception.o
Building object ../build/filemem.o
Building object ../build/fstring.o
Building object ../build/getopts.o
Building object ../build/histitem.o
Building object ../build/history.o
Building object ../build/file.o
Building object ../build/linuxterm.o
Building object ../build/terminal.o
Building library.
```

**Comment by 4ctrl-alt-del** [comment]
For the warnings you get about OpenCL, as per this link http://stackoverflow.com/questions/28500496/opencl-function-found-deprecated-by-visual-studio a couple opencl 1.2 functions where apparently deprecated in opencl 2.0 but it still supports the old API, like the link says all we need to do is include a macro define at the beginning of the source's header.

**Comment by 4ctrl-alt-del** [comment]
Fixed in version 1.2.0 release.
